File: Python/Algorithms/RecursionAlgorithms/FibonacciNumber.py
"""
__FileCreationDate__  :  4/5/2020
__Author__           :  CodePerfectPlus
__Package__         :  Python 3
__GitHub__         : https://www.github.com/codeperfectplus
"""


def fibonacciNumber(n):
    """
    :return: n'th fibonacci Number
    """
    if n == 0:
        return 0
    elif n == 1:
        return 1
    else:
        return fibonacciNumber(n - 1) + fibonacciNumber(n - 2)


print(fibonacciNumber(3))

File: C++/Algorithms/RecursionAlgorithms/FibonacciNumber.cpp
#include <iostream>
using namespace std;

int fibonacciNumber(int n){
    // Checks if n<=1, i.e. 0 and 1
    if(n<=1) {
        return n;
    }
    else {
        // Returns the function recursively for the next number 
        return fibonacciNumber(n - 1) + fibonacciNumber(n - 2);
    }
}

int main() {
    
    /* 
        Fibonacci Numbers Algorithm Implementation in C++
        
        Output : Return the element at the index 'i' in a Fibonacci series.
        Time complexity : O(n)

        Example:
        i=3

        Output:
        2
    */

    int i=3;
    cout << fibonacciNumber(i)<<endl;
}